src/job_shop_scheduler_cqm.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2023 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import dimod
+
+from dwave.system import LeapHybridCQMSampler
+
+from src.model_data import JobShopData
+
+
+def sum_to_one(*args):
+    return sum(args) == 1
+
+
+def get_label(task, time):
+    """Creates a standardized name for variables in the constraint satisfaction problem,
+    JobShopScheduler.csp.
+    """
+    return "{task.job}_{task.resource},{time}".format(**locals())
+
+
+class JobShopScheduler:
+    def __init__(self, model_data: JobShopData):
+        """
+        Args:
+            model_data: A JobShopData object that holds the data for this job shop 
+            scheduling problem.
+
+        """
+        self.model_data = model_data
+
+        self._process_data()
+        self._initialize_variables()
+        self.cqm = dimod.ConstrainedQuadraticModel()
+        self._add_task_start_constraint()
+        self._add_precedence_constraint()
+        self._add_share_machine_constraint()
+        self._remove_absurd_times()
+        # self._model_end()
+        self._add_objective()
+
+
+    def solve(self, profile: str=None) -> dict:
+        """Solves the job shop scheduling problem.
+        
+        Args:
+            profile: A string. The name of the profile to use for the hybrid solver.
+
+        Returns:
+            dict: A dictionary of the form {task: (start_time, end_time)} for each task in the
+            problem.
+        """
+        sampler = LeapHybridCQMSampler(profile=profile)
```

Comment:
Might want to make this a class-level object. It will make the round trip time for small problems a bit faster by skipping the solver connection step.